test/nsidc-smap/nsidc-smap_Regression.ipynb

```diff
@@ -0,0 +1,354 @@
+    "            for shortname, test_config in test_configs.items():\n",
+    "                print(f\"running request: {shortname}\")\n",
+    "\n",
+    "                test_request = Request(**test_config[\"request_params\"])\n",
+    "                ext = test_config[\"test_params\"][\"ext\"]\n",
+    "                test_output = tmp_dir / Path(f\"{shortname}_{test_name}{ext}\")\n",
+    "                test_reference = Path(\n",
+    "                    f\"reference_files/{test_output.stem}_reference.json\"\n",
+    "                )\n",
+    "\n",
+    "                submit_and_download(harmony_client, test_request, test_output)\n",
+    "\n",
+    "                assert exists(\n",
+    "                    test_output\n",
+    "                ), f\"Unsuccessful Harmony Request: {shortname}: {test_name}\"\n",
+    "\n",
+    "                # Grab correct comparison function and call it.\n",
+    "                compare_fxn = comparison_function_by_extension(ext)\n",
+    "                reference_file = (\n",
+    "                    Path(\"reference_files\") / f\"{test_output.stem}_reference.json\"\n",
+    "                )\n",
+    "                print(\n",
+    "                    f\"assert: {compare_fxn.__name__}({test_output.name}, {reference_file}) \"\n",
+    "                )\n",
+    "                assert compare_fxn(test_output, reference_file)\n",
+    "\n",
+    "                print_success(f\"{shortname} {test_name} test request complete.\")\n",
+    "            print_success(f\"{test_name} test suite complete.\")\n",
+    "        print_success(\"Entire SMAP Single Output Regression Test suite complete.\")\n",
+    "else:\n",
+    "    print(\n",
+    "        f'Bounding box tests not configured for environment: {configuration[\"env\"]} - skipping tests'\n",
+    "    )"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "80ba1a0f-6928-4906-9002-f0a15c75f381",
+   "metadata": {},
+   "source": [
+    "## Multiple output file tests\n",
+    "\n",
+    "The following cell runs through any request that generates multiple output files. These are GeoTIFF Reformatting requests. These are generally requests for multiple variables and generate a single GeoTIFF for each variable on the output. The tests make a request and then download all of the resulting files and then compare those to the reference hash values to verify no changes in the output.\n",
+    "\n",
+    "The tests include:\n",
+    "\n",
+    "- SPL2SMP with multiple variable subsetting.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "6e5e2c1c-4ad6-4761-b001-1a90af7c1bf7",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "if configuration is not None:\n",
+    "    with TemporaryDirectory() as tmp_dir:\n",
+    "        for test_name, test_configs in configuration[\"multiple_output_tests\"].items():\n",
+    "\n",
+    "            for shortname, test_config in test_configs.items():\n",
```

Comment:
Same comment on the ordering of the loops and the declaration of a `TemporaryDirectory` (in fairness, I guess all the outputs aren't going in the same directories - but they are only being split across two).

Reply:
57b838a